docs/migration.md

```diff
@@ -0,0 +1,299 @@
+# Garden v8 Migration Guide
```

Comment:
This is going to be a continuous migration guide, not just for v8. We may want to make the title `Garden migration guide` and have an `h2` as `v8`. All the other headings should be incremented as well.